compiler/test/test_end_to_end.re

```diff
@@ -1283,47 +1284,83 @@ let string_tests = {
     tparse(
       "string_parse_dqs1",
       "\"foo\"",
-      {statements: [str("foo")], prog_loc: Location.dummy_loc},
+      {
```

Comment:
@ospencer Maybe add a test case?

Reply:
Good call here. I am trying to write these now and it seems like a bunch of comments are being added to this block.

Investigating.

Reply:
I added a test for this in a separate block (see individual commit).